src/xclim/indices/generic.py

```diff
@@ -839,6 +846,42 @@ def season(
     return out
 
 
+def season_length_from_boundaries(season_start: xr.DataArray, season_end: xr.DataArray) -> xr.DataArray:
+    """
+    Season length using pre-computed boundaries.
+
+    Parameters
+    ----------
+    season_start : xr.DataArray
+        Day of year where the season starts.
+    season_end : xr.DataArray
+        Day of year where the season ends.
+
+    Returns
+    -------
+    xr.DataArray, [dimensionless]
+        Length of the season.
+
+    Notes
+    -----
+    If `season_start` and `season_end` are computed with different resampling frequencies, the time
+    of `season_start` are selected to write the output. This is useful when season start and end were computed
+    with different resampling frequencies. Otherwise, functions in ``xclim.indices.run_length`` will be appropriate.
+    """
+    days_since_start = doy_to_days_since(season_start, start=dayofyearstr_from_freq(xr.infer_freq(season_start.time)))
```

Comment:
Wait, isn't the default behaviour with `start=None` exactly what you're doing here ?

Also, to get the `DayOfYearStr` your essentially doing : `date_range(..., freq=infer_freq(...))`, isn't that running in circles ?